lib/tests/formulae.rb

```diff
@@ -235,28 +239,18 @@ def formula!(formula_name, args:)
           skipped formula_name, "#{formula.full_name} has been disabled!"
           return
         end
-        new_formula = @added_formulae.include?(formula_name)
 
-        if Hardware::CPU.arm? &&
-           args.skip_unbottled_arm? &&
-           !formula.bottled? &&
-           !formula.bottle_unneeded? &&
-           !new_formula
-          skipped formula_name, "#{formula.full_name} has not yet been bottled on ARM!"
-          return
+        all_deps_have_compatible_bottles = formula.deps.all? do |dep|
+          bottled?(dep.to_formula, no_older_versions: false)
```

Comment:
Having a `no_older_versions: false` double-negative seems unnecessarily confusing.

Reply:
Indeed, though this is helped by it being an optional parameter to `BottleSpecification#tag?` so we only ever specify it in `brew` if we are passing `true`.